crates/ruff_linter/src/rules/flake8_pie/rules/reimplemented_container_builtin.rs

```diff
@@ -31,46 +32,50 @@ use crate::checkers::ast::Checker;
 /// @dataclass
 /// class Foo:
 ///     bar: list[int] = field(default_factory=list)
+///     baz: dict[str, int] = field(default_factory=dict)
 /// ```
 ///
 /// ## References
 /// - [Python documentation: `list`](https://docs.python.org/3/library/functions.html#func-list)
 #[violation]
-pub struct ReimplementedListBuiltin;
+pub struct ReimplementedContainerBuiltin;
```

Comment:
Maybe just `ReimplementedBuiltin` would suffice?

Reply:
I think we have some other rules that also detect re-implemented builtins though -- like the simplify rules that detect re-implemented `any` and `all` loops.

Reply:
Hm yes `ReimplementedBuiltin` is used for `SIM110` and `SIM111` — that's such a generic name for those though...

I think `ReimplementedContainerBuiltin` is fine but separately we should consider changing that other one to be more specific :D